src/librbd/plugin/Api.h

```diff
@@ -29,6 +38,45 @@ struct Api {
       librados::snap_t snap_id, const ZTracer::Trace &trace,
       Context* on_finish);
 
+  virtual void execute_image_metadata_set(
+      ImageCtxT *image_ctx,
+      const std::string &key,
+      const std::string &value,
+      Context *on_finish);
+
+  virtual void execute_image_metadata_remove(
+      ImageCtxT *image_ctx,
+      const std::string &key,
+      Context *on_finish);
+
+  virtual std::string unique_lock_name(
```

Comment:
Nit: you define this but use a new `pwl::unique_lock_name` method?

Reply:
Removed this interface from Api.